Autopilot/AttitudeManager/PWM.h

```diff
@@ -1,150 +1,96 @@
- * Function: void initPWM(char inputChannels, char outputChannels);
- *
- * Preconditions: None.
- *
- * Overview: Initializes the PWM ports. For instance if inputChannels = 0b00100101
- *      then channels 1,3, and 6 will be initialized.
- *
- * Input: char inputChannels -> The inputs to be initialized. Each bit represents
- *          a channel.
- *
- * Output: None.
- *
- *****************************************************************************/
-void initPWM(char inputChannels, char outputChannels);
-
-/*****************************************************************************
- * Function: void PWMInputCalibration(unsigned int channel, float signalScaleFactor, unsigned int signalOffset);
- *
- * Preconditions: None.
- *
- * Overview: Calibrates the input range and trim of the device.
- *
- * Input: unsigned int channel -> The channel to calibrate.
- *        float signalScaleFactor -> The scale factor to increase or decrease the range by.
- *                                  For instance, if the target range is from -1024 to 1024,
- *                                  the scale factor would be 1024/(half the detected input range)
- *        unsigned int signalOffset -> Compensation for a non-zero offset on the input. This is
- *                                      known as trim.
- *
- * Output: None.
- *
- *****************************************************************************/
-void PWMInputCalibration(unsigned int channel, float signalScaleFactor, unsigned int signalOffset);
+/**
+ * Maximum and minimum limits received from the controller. These should be calibrated,
+ * and the scaling from these values to the MAX and MIN PWM values will depend on these.
+ * Note that these will be used as the default/initial values, however the actual scaling values
+ * can be set by the ground station
+ */
+#define UPPER_PWM 1284
+#define LOWER_PWM 642
 
-/*****************************************************************************
- * Function: void PWMdOutputCalibration(unsigned int channel, float signalScaleFactor, unsigned int signalOffset);
- *
- * Preconditions: None.
- *
- * Overview: Calibrates the output range and trim of the device.
- *
- * Input: unsigned int channel -> The channel to calibrate.
- *        float signalScaleFactor -> The scale factor to increase or decrease the range by.
- *                                  For instance, if the range is from -1024 to 1024,
- *                                  the scale factor would be (half the desired output range)/1024
- *        unsigned int signalOffset -> Compensation for a non-zero offset on the input. This is
- *                                      known as trim.
- *
- * Output: None.
- *
- *****************************************************************************/
-void PWMOutputCalibration(unsigned int channel, float signalScaleFactor, unsigned int signalOffset);
+#define SP_RANGE MAX_PWM//(UPPER_PWM - MIDDLE_PWM)
```

Comment:
Its supposed to just be a scaling factor for half the PWM range. Sort of similar to how you typically convert between deg and rad using 180/pi.

I guess an appropriate definition would be the PWM half range.
Which in terms of readability is better than just "MAX_PWM".

The actual appropriate definition should probably be: (MAX_PWM - MIN_PWM)/2
Which in this case (and probably all cases) is equal to 1024.